synapse/rest/client/sync.py

```diff
@@ -301,14 +301,13 @@ async def encode_response(
         if archived:
             response["rooms"][Membership.LEAVE] = archived
 
-        # By the time we get here groups is no longer optional.
-        assert sync_result.groups is not None
```

Comment:
I double checked this:

`_generate_sync_entry_for_groups` was computing this and making it non-None. But we're not calling that function any more so it'll continue to be None.

(I'm not quite sure why it `.groups` was allowed to be None in the first place? But it's going away soon anyway.)

Reply:
Thanks for double check, that was what I found as well! 👍 (I also ran some sync tests after disabling groups and nothing 💥 ...)